test/t8_forest_incomplete/t8_gtest_iterate_replace.cxx

```diff
@@ -249,11 +246,9 @@ TEST_P (forest_iterate, test_iterate_replace)
 
     t8_forest_iterate_replace (forest_adapt, forest, t8_forest_replace);
     t8_forest_unref (&forest);
-    if (runs > 1) {
-      /* Partition the forest. This is useful,
-      * if we run the test a second time with the adapted forest. */
-      forest_adapt = t8_adapt_forest (forest_adapt, NULL, 0, 1, NULL);
-    }
+
+    /* Partition the forest. This is useful as preparation for the second run with the adapted forest. */
+    forest_adapt = t8_adapt_forest (forest_adapt, NULL, 0, 1, NULL);
```

Comment:
I know that this has nothing to do with your PR, but is this really the cleanest way available to partition a forest in t8code?

Reply:
I am not sure but as we need an uncomitted forest to call the function t8_forest_set_partition, it is maybe the best way to just use this function t8_adapt_forest, which is implemented in this test file. (?)

Reply:
I was not aware that this function is just implemented in this testfile.

I think it would be nice to have separate wrappers around the highlevel functionality in the public interface, like adapt, partition, etc., that take a commited forest, setup the new forest internally with all needed parameters, and return the commited forest.

But this is a separate issue.